File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int evenCount = 0;
        
        for(int i = 0; i < nums.length; i++){
            int numbersCount = 0;

            while(nums[i] > 0){
                nums[i] /= 10;
                numbersCount++;
            }
            if(numbersCount % 2 == 0){
                evenCount++;       
            }
        }

        return evenCount;
    }
}

File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.kt
class Solution {
    fun findNumbers(nums: IntArray): Int {

        var evenCounts = 0

        for(num in nums) {
            var numberCount = 0
            var currNumber = num

            while(currNumber > 0) {
                numberCount++
                currNumber /= 10
            }
            if(numberCount % 2 == 0) evenCounts++
        }

        return evenCounts
    }
}